pkg/controllers/edgeconnect/controller.go

```diff
@@ -174,3 +259,243 @@ func (controller *Controller) reconcileEdgeConnect(edgeConnect *edgeconnectv1alp
 	}
 	return nil
 }
+
+func (controller *Controller) reconcileEdgeConnectProvisioner(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) error { //nolint: revive
+	log.Info("reconcileEdgeConnectProvisioner")
+
+	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
+	if err != nil {
+		return err
+	}
+
+	edgeConnectId, err := getEdgeConnectIdByName(edgeConnectClient, edgeConnect.Name)
+	if err != nil {
+		return err
+	}
+
+	edgeConnectIdFromSecret, err := controller.getEdgeConnectIdFromClientSecret(ctx, edgeConnect)
+	if err != nil {
+		return err
+	}
+
+	if edgeConnectId != "" {
+		if edgeConnectIdFromSecret == "" {
+			log.Info("EdgeConnect has to be recreated due to missing secret")
+			// TODO: managedByDynatraceOperator attribute needed
+			if err := edgeConnectClient.DeleteEdgeConnect(edgeConnectId); err != nil {
+				return err
+			}
+			edgeConnectId = ""
+		} else if edgeConnectId != edgeConnectIdFromSecret {
+			// TODO: managedByDynatraceOperator attribute needed
+			log.Info("EdgeConnect has to be recreated due to invalid Id")
+			if err := edgeConnectClient.DeleteEdgeConnect(edgeConnectId); err != nil {
+				return err
+			}
+			edgeConnectId = ""
+		}
+	}
```

Comment:
is this possible, considering we have the `managedByDynatraceOperator` ?

if not, then we should make it clear that this logic can be removed when we start using `managedByDynatraceOperator`

Reply:
offline conversation summary: there should be an extra if there when we have the new property